content/news/2017-02-20-mapstruct-1_2_0_Beta1-is-out-with-lombok-support-and-direct-field-access.md

```diff
@@ -0,0 +1,195 @@
+Thanks a lot everyone for all your hard work!
+
+Enough of the pep talk, let's take a closer look at some of the new features!
+
+### Using MapStruct together with Project Lombok
+
+Being able to use MapStruct together with [Project Lombok](https://projectlombok.org/) within a single compilation unit has been a long-awaited and [intensely debated](https://github.com/mapstruct/mapstruct/issues/510) feature request.
+
+The challenge here was that Lombok actually _alters_ the AST of the compiled classes (something which has never been foreseen by the [annotation processing](https://www.jcp.org/en/jsr/detail?id=269) infrastructure).
+Other annotation processors such as MapStruct then wouldn't know about the elements added by Lombok (e.g. getters and setters for the `@Data` annotation).
+
+To cut a long story short, thanks to a very productive collaboration with the Lombok team this could be resolved.
+MapStruct will now wait until Lombok has done all its amendments before generating mapper classes for Lombok-enhanced beans.
+Note that Lombok 1.16.14 or newer is required.
+
+### Java 8 stream support
+
+Java 8's `Stream` type can [be used](https://github.com/mapstruct/mapstruct/issues/962) now as source and target of mapping methods:
+
+{{< prettify xml >}}
+@Mapper
+public interface OrderMapper {
+
+    Set<OrderDto> ordersToOrderDtos(Stream<Order> orders);
+
+    OrderDto orderToOrderDto(Order order);
+}
+{{< /prettify >}}
+
+If the "element type" differs between source and target stream/collection (`Order` vs. `OrderDto`),
+`Stream#map()` will be used to call the right element mapping method (`orderToOrderDto()`) in the generated code.
+
+### Mappings using direct field access
+
+Traditionally MapStruct relied on the JavaBeans convention and called getter and setter methods for propagating property values from source to target.
+Sometimes writing all the getters and setters may be a bit cumbersome, though, e.g. when it comes to "dumb" DTOs which should just hold a couple of property values.
+
+Using public fields is perfectly fine in such case, which is why direct field access is [supported by MapStruct now](https://github.com/mapstruct/mapstruct/issues/557).
+I.e. if your source or target beans do not define getters or setters for the properties but declare public fields,
+the generated code will directly access the fields.
+Note that no reflection will be used, i.e. we won't make private fields accessible.
+
+You can find a small example demonstrating field-based mapping [here](https://github.com/mapstruct/mapstruct-examples/tree/master/mapstruct-field-mapping/).
+
+### Automatic creation of sub-mapping methods
```

Comment:
Maybe you want to add a link to the mapstruct-nested-bean-mappings - https://github.com/mapstruct/mapstruct-examples/tree/master/mapstruct-nested-bean-mappings. The link is not active yet because the PR is not merged yet :)